src/saveload/afterload.cpp

```diff
@@ -1017,6 +1017,27 @@ bool AfterLoadGame()
 		_settings_game.construction.freeform_edges = false;
 	}
 
+	/* Handle infinite water borders. */
+	if (IsSavegameVersionBefore(SLV_INFINITE_WATER_BORDERS)) {
+		if (CheckWaterBorders(true)) {
+			/* We passed the water borders check, yay! */
+			_settings_game.game_creation.water_borders = BORDERFLAGS_ALL;
+
+			/* Flatten void tiles, which may have been affected by terraforming near the map edge. */
+			for (uint x = 0; x <= Map::SizeX(); x++) {
+				SetTileHeightOutsideMap(x, 0, 0);
+				SetTileHeightOutsideMap(x, Map::SizeY(), 0);
+			}
+			for (uint y = 1; y < Map::SizeY(); y++) {
+				SetTileHeightOutsideMap(0, y, 0);
+				SetTileHeightOutsideMap(Map::SizeY(), y, 0);
+			}
```

Comment:
Wouldn't the bottom edges should always already be at 0, otherwise it wouldn't be fully-watered edge?

Reply:
If I'm reading this properly (and understanding void tiles properly), `SizeX()` is a void tile while `MaxX()` is a real tile that is watered (and likewise for Y).

I did make a mistake with this code, the second should be `SizeX(), y` and not `SizeY(), y`. Fixed now. 🙂